depthai_bridge/src/CameraControl.cpp

```diff
@@ -0,0 +1,172 @@
+    controlQueue->send(ctrl);
+}
+
+void CameraControl::setDevice(std::shared_ptr<dai::Device> device) {
+    _device = device;
+}
+
+dai::CameraControl::AutoFocusMode CameraControl::getFocusMode() {
+    if(_focus.mode == "AUTO") return dai::CameraControl::AutoFocusMode::AUTO;
+    if(_focus.mode == "CONTINUOUS_PICTURE") return dai::CameraControl::AutoFocusMode::CONTINUOUS_PICTURE;
+    if(_focus.mode == "CONTINUOUS_VIDEO ") return dai::CameraControl::AutoFocusMode::CONTINUOUS_VIDEO;
+    if(_focus.mode == "EDOF") return dai::CameraControl::AutoFocusMode::EDOF;
+    if(_focus.mode == "MACRO") return dai::CameraControl::AutoFocusMode::MACRO;
+    if(_focus.mode == "OFF") return dai::CameraControl::AutoFocusMode::OFF;
+    return dai::CameraControl::AutoFocusMode::AUTO;
+}
+
+int clamp(int v, int min, int max) {
+    if(v < min) return min;
+    if(v > max) return max;
+    return v;
+}
+
+void CameraControl::setFocus() {
+    dai::CameraControl ctrl;
+    auto controlQueue = _device->getInputQueue("control");
+    auto mode = getFocusMode();
+    ctrl.setAutoFocusMode(mode);
+
+    if(_focus.mode == "OFF") {
+        auto configQueue = _device->getInputQueue("config");
+        dai::ImageManipConfig cfg;
+        cfg.setCropRect(_focus.region.at(0), _focus.region.at(1), 0, 0);
+        configQueue->send(cfg);
+    } else if(_focus.region.at(2) > 0 && _focus.region.at(3) > 0) {
+        ctrl.setAutoFocusRegion(_focus.region.at(0), _focus.region.at(1), _focus.region.at(2), _focus.region.at(3));
+    } else if(_focus.mode != "AUTO") {
+        ctrl.setManualFocus(clamp(_lens_position, 0, 255));
+    }
+
+    controlQueue->send(ctrl);
+}
+
+req_type CameraControl::setRgbExposureRequest(exp_req_msg request, exp_rep_msg response) {
+    _rgb.auto_exposure = req_get(auto_exposure);
+    _rgb.region.at(0) = req_get(exposure_x);
+    _rgb.region.at(1) = req_get(exposure_y);
+    _rgb.region.at(2) = req_get(exposure_width);
+    _rgb.region.at(3) = req_get(exposure_height);
+    _rgb.compensation = req_get(compensation);
+    _rgb.time_us = clamp(req_get(exposure_time_us), 1, 33000);
+    _rgb.sensitivity_iso = clamp(req_get(exposure_time_us), 1, 33000);
+    setExposure(_rgb);
+    rep_get(success) = true;
+    bool result = true;
+    return (req_type)result;
+}
+
+req_type CameraControl::setStereoExposureRequest(exp_req_msg request, exp_rep_msg response) {
+    _stereo.auto_exposure = req_get(auto_exposure);
+    _stereo.region.at(0) = req_get(exposure_x);
+    _stereo.region.at(1) = req_get(exposure_y);
+    _stereo.region.at(2) = req_get(exposure_width);
+    _stereo.region.at(3) = req_get(exposure_height);
+    _stereo.compensation = req_get(compensation);
+    _stereo.time_us = clamp(req_get(exposure_time_us), 1, 33000);
+    _stereo.sensitivity_iso = clamp(req_get(exposure_time_us), 1, 33000);
+    setExposure(_stereo);
+    rep_get(success) = true;
+    bool result = true;
+    return (req_type)result;
+}
```

Comment:
Here In this case since you are saving those to a member variables you have to write two functions. 
What do we do when we have a device with 4 cameras ? 
Or two RGB cameras ? 
Or if the user is not running stereo but running only left and right  camera ? 

You will keep adding new functions and new local variables. Avoid that. 

There are Multiple ways you can do this. 
1. Remove this stereoExposureRequest function. Have only one. And for each camera created in the node create one CameraControl Object and use it only for that camera. In stereo inertial Case you will have one for RGB if RGB is on and One for Stereo (AKA Left. SInce right camera is synced to left). 
2. Use local variables and have an input on which queue from the service (I don't like this idea)